server/controllers/User.js

```diff
@@ -136,34 +145,50 @@ class UserController {
    * @returns {Object} the response body
    */
   static update(req, res, next) {
-    User.findById(req.payload.id)
+    const { decoded } = req.body;
+    // If req.body.user is defined, de-structure it to get the parameters
+    // else, set the parameters to undefined
+    const {
+      firstName, lastName, username, password, bio, image,
+    } = req.body.user ? req.body.user
+      : {
+        firstName: undefined,
+        lastName: undefined,
+        username: undefined,
+        password: undefined,
+        bio: undefined,
+        image: undefined,
```

Comment:
You can avoid the extra work of creating this object
```
: {
        firstName: undefined,
        lastName: undefined,
        username: undefined,
        password: undefined,
        bio: undefined,
        image: undefined,
};
```
and also the database operation can be avoided if `req.body.user` is not defined by using an `if` statement to check before it get there. That is you can stop the operations and/or return an error if not defined.